index.html

```diff
@@ -1421,6 +1421,104 @@ <h3>Semantic Interoperability</h3>
+<a>property</a>:
+        </p>
+
+        <dl>
+          <dt><var>credentialSchema</var></dt>
+          <dd>The value of this property MUST be one or more data schemas
+            that provide <a>verifiers</a> with enough information to
+            determine if the provided data is conformant to the provided schema.
+            Each <code>credentialSchema</code> comprises at least its
+            <code>type</code> (e.g. <code>JsonSchemaValidator2018</code>),
+            and a <code>id</code> property that MUST be a URI identifying
+            the schema file. The precise contents of each
+            data schema is determined by the specific type definition.
+          </dd>
+        </dl>
+
+        <pre class="example nohighlight" title="Usage of credentialSchema property to do JSON Schema validation">
+{
+  "@context": [
+    "https://w3.org/2018/credentials/v1",
+    "https://example.org/motorlicense/v1"
+  ],
+  "id": "http://dmv.example.gov/credentials/3732",
+  "type": ["VerifiableCredential", "ProofOfAgeCredential"],
+  "issuer": "https://dmv.example.gov/issuers/14",
+  "issuanceDate": "2010-01-01T19:73:24Z",
+  "claim": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "ageOver": 21
+  },
+  <span class="highlight">"credentialSchema": {
+    "id": "https://example.org/motorlicense/proof-of-age.json",
+    "type": "JsonSchemaValidator2018"
+  }</span>,
+  "proof": { ... }
+}
+        </pre>
+
+        <p>
+In the example above, the <a>issuer</a> is specifying a
+<code>credentialSchema</code> that points to a JSON Schema file that can be
+used by <a>verifier</a> to determine if the <a>verifiable credential</a> is
+well-formed.
+        </p>
+
+        <p>
+Data schemas can also be used to specify mappings to other binary formats,
+such as those used to perform zero-knowledge proofs.
+        </p>
+
+        <pre class="example nohighlight" title="Usage of credentialSchema property to do zero-knowledge validation">
+{
+  "@context": [
+    "https://w3.org/2018/credentials/v1",
+    "https://example.org/motorlicense/v1"
+  ],
+  "id": "http://dmv.example.gov/credentials/3732",
+  "type": ["VerifiableCredential", "ProofOfAgeCredential"],
+  "issuer": "https://dmv.example.gov/issuers/14",
+  "issuanceDate": "2010-01-01T19:73:24Z",
+  "claim": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "ageOver": 21
+  },
+  <span class="highlight">"credentialSchema": {
+    "id": "https://example.org/motorlicense/proof-of-age.zkp",
+    "type": "ZkpExampleSchema2018"
+  }</span>,
+  "proof": { ... }
+}
+        </pre>
+
+        <p>
+In the example above, the <a>issuer</a> is specifying a
+<code>credentialSchema</code> that points to a zero-knowledge packed binary
+data format that is capable of transforming the input data into a format
+that can then be used by a <a>verifier</a> to determine if the
+proof provided with the <a>verifiable credential</a> is valid.
```

Comment:
This section might need a little tweaking to make it clear if we're looking at the verifiable credential that's directly from the issuer or not.

If we are, then the text needs to be tweaked to avoid sounding like the `proof` here is being presented when, in fact, a whole new credential must be derived and presented. I imagine, in this case, that the holder/presenter would need to include the `credentialSchema` in a derived credential in the presentation such that the verifier could retrieve it from a registry and confirm its validity... and then use it to encode the data presented in the derived credential so that the ZKP-based proof (on the derived credential) could be verified.

If we're not, we shouldn't have things like the credential ID in there and so on.

@brentzundel ... could you take a look at this language here and see if it needs tweaking and propose something that works?

Reply:
It may need tweaking.
The notion of a data schema seems very similar to the objects that @ken-ebert and I have been working on, which we will be discussing next Monday. 
I propose we wait to merge this PR until that conversation, after which I hope we will all be on the same page.

Reply:
Ok, on hold until after the ZKP call.

Reply:
I'm going to add an issue, merge this, and then modify it based on the end result of the ZKP discussion.